bilalamjad0x01/news-app
# ui done

# News App

This PR represents the culmination of development efforts for the "THE BALL STREET JOURNAL" News App. The app fetches and displays news articles, allowing users to view details and bookmark their favorite stories. UI, UX, and backend logic have all been implemented to deliver a seamless news reading experience.

## Key Features:

- **Home Screen:** Displays the latest news articles with a special highlight for the top story.
- **Article Details:** Users can click on an article to read it in detail.
- **Bookmarks:** Articles can be saved for later reading in the Bookmarks tab.
- **Share:** Users can click on share button and native share functionality will appear with the url of article copied.
- **Theming:** Users have the ability to switch between light and dark themes using the floating action button. Shared_preferences used for persistent theming.
- **SafeArea Implementation:** Ensured UI components respect device notches and system UI elements.
- **Custom Fonts & Styling:** Introduced custom fonts and styles to enhance the visual appeal.

## How to test:

1. Install the app and launch it.
2. Browse through the articles displayed on the home screen.
3. Open an article to view its details.
4. Use the bookmark feature to save favorite articles.
5. Switch between light and dark themes using the floating action button.
6. Test on various devices to ensure consistent layout and functionality.

## Screenshots:

**Light & Dark modes:**  
<table>
  <tr>
    <td valign="top">
      <br>
      <img src="https://github.com/bilalamjad0x01/news-app/assets/66705229/4960ecc5-52f8-4520-88b8-7dc2a22c0135" alt="lightMode">
    </td>
    <td valign="top">
      <br>
      <img src="https://github.com/bilalamjad0x01/news-app/assets/66705229/8134a386-0741-4ee4-8948-d23df4c5ab01" alt="darkMode">
    </td>
  </tr>
</table>


**App in Action:**  

![VID_20230815103543_AdobeExpress](https://github.com/bilalamjad0x01/news-app/assets/66705229/4cf9ac03-d26b-4622-a2a2-8195ef36a9b6)
